### ISSUE TITLE
[plugin] [opds browser] [bug] gutenberg downloads fail due to books being moved; 302 redirect html downloads instead

### ISSUE THREAD

**Issue by arooni** [koreader/koreader#6149]
* KOReader version: 2020 04 01
* Device: PW4

#### Issue
Project Gutenberg is apparently sending 302 messages for any download saying the download has been moved.  This seems to be the case for any download.

```
└─(0.77:67%:20:50:79%:#)── cat Shelley_\ Mary\ Wollstonecraft\ -\ Frankenstein\;\ Or_\ The\ Modern\ Prometheus.epub                         ──(Thu, May 14)─┘
<!DOCTYPE HTML PUBLIC "-//IETF//DTD HTML 2.0//EN">
<html><head>
<title>302 Found</title>
</head><body>
<h1>Found</h1>
<p>The document has moved <a href="http://www.gutenberg.org/cache/epub/84/pg84.epub">here</a>.</p>
<hr>
<address>Apache Server at www.gutenberg.org Port 80</address>
</body></html>
```

**Comment by Abish4i** [comment]
### I have same issue though PDF files are getting write access but unable to save MOBI & EPUB files...

**Comment by Frenzie** [comment]
They redirect from https → http. I'm not sure if that's supposed to be supported.

```
$ curl -I https://www.gutenberg.org/ebooks/84.epub.noimages
HTTP/1.1 302 Found
Server: Apache
Location: http://www.gutenberg.org/cache/epub/84/pg84.epub
Content-Type: text/html; charset=iso-8859-1
X-Powered-By: 1
X-Cacheable: NO: beresp.status
Date: Thu, 25 Jun 2020 13:51:17 GMT
X-Varnish: 467323675
Age: 0
Via: 1.1 varnish
```

**Comment by Frenzie** [comment]
As I thought might be the case, while the OPDS browser is still using some methods that have thankfully become obsolete over the past year (see https://github.com/koreader/koreader-base/issues/594), this particular example is purposefully disabled regardless.

```lua
    -- avoid https downgrades
    if ('https' == reqt.scheme) and ('https' ~= scheme) then return false end
```

**Comment by poire-z** [comment]
Possibly solved by #6334. Re-open if not.
